mito-ai/src/utils/websocket/websocketClient.ts

```diff
@@ -134,17 +148,34 @@ export class CompletionWebsocketClient implements IDisposable {
   ): Promise<R> {
     return new Promise<R>(async (resolve, reject) => {
       try {
-        // Ensure the websocket is initialized before sending
-        await this.ready;
-        
-        const pendingReply = new PromiseDelegate<R>();
-        if (this._socket) {
-          this._socket.send(JSON.stringify(message));
+        // If the socket is not connected, try to reconnect first
+        if (this._socket === null || this._socket.readyState !== WebSocket.OPEN) {
+          try {
+            console.log('Connection is closed, attempting to reconnect before sending message...');
+            
+            // Reset the ready promise since we're going to reconnect
+            this._ready = new PromiseDelegate<void>();
+            
+            await this.reconnect();
+            console.log('Successfully reconnected, now sending message');
+          } catch (reconnectError) {
+            console.error('Failed to reconnect websocket:', reconnectError);
+            reject(new Error('Failed to reconnect websocket before sending message'));
+            return;
+          }
+        }
+
+        if (this._socket && this._socket.readyState === WebSocket.OPEN) {
+          const id = message.message_id ?? crypto.randomUUID();
```

Comment:
It looks like previously we were just sending `this._socket.send(JSON.stringify(message))` . Do we want to be creating a new id here if it doesn't exist?